src/libs/hovercard/hovercard.js

```diff
@@ -0,0 +1,184 @@
+			addHoverCard(nametag, id);
+		}, listenerTimeout);
+	});
+
+	nametag.addEventListener("mouseleave", () => {
+		hovering = null;
+		setTimeout(() => {
+			hovercardExists = false;
+			if (hovering === nametag) return;
+			removeHoverCard(layer.querySelector(".protoots-hovercard"));
+		}, listenerTimeout);
+	});
+}
+
+/**
+ *
+ * @param {HTMLElement} el Element which will be hovered on
+ * @param {*} statusID Status data-id
+ */
+async function addHoverCard(el, statusID) {
+	if (hovercardExists) return;
+
+	hovercardExists = true;
+
+	//remove all other hovercards
+	document.querySelectorAll("#protoots-hovercard").forEach((element) => element.remove());
+
+	//remove title of target element, to avoid double popup
+	el.setAttribute("title", "");
+
+	const hovercard = document.createElement("div");
+
+	const accountName = normaliseAccountName(el.querySelector(".display-name__account").textContent);
+
+	//show placeholder
+	const placeholder = document.createElement("img");
+	placeholder.src = runtime.getURL("icons/fennecspin.gif");
+
+	const tempHeader = createElementWithClass("div", "account__header");
+
+	tempHeader.appendChild(placeholder);
+	hovercard.appendChild(tempHeader);
+
+	//set position from mouse cursor
+	hovercard.style.left = (mousePos.x + 10).toString() + "px";
+	hovercard.style.top = (mousePos.y + 10).toString() + "px";
+
+	hovercard.classList.add("protoots-hovercard");
+	hovercard.id = "protoots-hovercard";
+
+	layer.appendChild(hovercard);
+
+	//get account
+	const { account, relationship } = await fetchProfile(statusID, accountName);
+	if (!account) return;
+
+	const settings = hoverCardSettings();
+
+	//generate the profile and add it to the card
+	const [profileElement, bio] = generateProfile(account, relationship, settings);
+	hovercard.appendChild(profileElement);
+
+	//delete placeholder
+	hovercard.removeChild(tempHeader);
+
+	//get relationship from API again and redo those elements
+	//this is because a cached relationship may easily be out of date
+	//i.e. the users sees someone new, clicks follow, which then isn't reflected in the card
+	const updatedRelationship = await fetchRelationship(account.id);
+	const [updatedProfileElement, updatedbio] = generateProfile(
+		account,
+		updatedRelationship,
+		settings,
+	);
+	hovercard.replaceChild(updatedProfileElement, profileElement);
+
+	//If bio would be scrollable, but that settings is disabled, add the button
+	if (updatedbio.scrollHeight > updatedbio.clientHeight && !settings.scrollableBio) {
+		// console.log("bio is scrollable");
+		// bio.style.overflow = "hidden";
+		updatedbio.classList.add("showmore");
+		addShowMoreButton(updatedbio, hovercard);
+	}
+
+	moveOnScreen(hovercard);
+
+	hovercard.addEventListener("mouseenter", () => (hovering = el));
+	hovercard.addEventListener("mouseleave", () => {
+		hovering = false;
+		hovercardExists = false;
+		setTimeout(() => {
+			if (hovering) return;
+			removeHoverCard(hovercard);
+		}, listenerTimeout);
```

Comment:
see above for hovercardExists, and why is it here outside of setTimeout? also hovering was set to null, not false in the other event listener.
seems fishy/inconsistent.